model/textparse/protobufparse.go

```diff
@@ -160,13 +160,41 @@ func (p *ProtobufParser) Histogram() ([]byte, *int64, *histogram.Histogram) {
 		sh.NegativeSpans[i].Offset = span.GetOffset()
 		sh.NegativeSpans[i].Length = span.GetLength()
 	}
+
+	if isFloatHistogram(h) {
+		fh := histogram.FloatHistogram{
+			Count:           h.GetSampleCountFloat(),
+			Sum:             sh.Sum,
+			ZeroThreshold:   sh.ZeroThreshold,
+			ZeroCount:       h.GetZeroCountFloat(),
+			Schema:          sh.Schema,
+			PositiveSpans:   sh.Copy().PositiveSpans,
+			PositiveBuckets: castInt64ToFloat64(sh.PositiveBuckets),
+			NegativeSpans:   sh.Copy().NegativeSpans,
+			NegativeBuckets: castInt64ToFloat64(sh.NegativeBuckets),
+		}
+		if ts != 0 {
+			return p.metricBytes.Bytes(), &ts, nil, &fh
+		}
+		// Nasty hack: Assume that ts==0 means no timestamp. That's not true in
+		// general, but proto3 has no distinction between unset and
+		// default. Need to avoid in the final format.
+		return p.metricBytes.Bytes(), nil, nil, &fh
+	}
+
 	if ts != 0 {
-		return p.metricBytes.Bytes(), &ts, &sh
+		return p.metricBytes.Bytes(), &ts, &sh, nil
 	}
-	// Nasty hack: Assume that ts==0 means no timestamp. That's not true in
-	// general, but proto3 has no distinction between unset and
-	// default. Need to avoid in the final format.
-	return p.metricBytes.Bytes(), nil, &sh
+	return p.metricBytes.Bytes(), nil, &sh, nil
+}
+
+// cast array of int64 to array of float64
+func castInt64ToFloat64(a []int64) []float64 {
```

Comment:
I'm not happy with the name `castInt64ToFloat64`, but I think we don't need this helper function anymore if @codesome 's comments are addressed.